src/citrine/resources/data_concepts.py

```diff
@@ -85,13 +95,21 @@ def from_dict(cls, d: dict):
         obj = super().from_dict(d)
 
         for field, clazz in cls.client_specific_fields.items():
-            if popped[field] is None:
-                setattr(obj, "_{}".format(field), None)
-            elif isinstance(popped[field], dict):
-                setattr(obj, "_{}".format(field), clazz.build(popped[field]))
+            value = popped[field]
+            if value is None:
+                deserialized = None
+            elif issubclass(clazz, DictSerializable):
+                if not isinstance(value, dict):
+                    raise TypeError(
+                        "{} must be a dictionary or None but was {}".format(field, value))
+                deserialized = clazz.build(value)
+            elif issubclass(clazz, Property):
```

Comment:
To avoid the namespace collision with the property attribute, could we use this as `properties.Property` (as we have in some other places?)

Reply:
That is actually a python syntax error, surprisingly. I could give it an alias though.

Reply:
anything to disambiguate it